CreateService/handler.ts

```diff
@@ -42,32 +49,54 @@ import {
   retrievedServiceToApiService
 } from "../utils/conversions";
 import { ServicePayloadMiddleware } from "../utils/middlewares/service";
+import { UpsertServiceEvent } from "../utils/UpsertServiceEvent";
 
 type ICreateServiceHandler = (
   context: Context,
   auth: IAzureApiAuthorization,
   clientIp: ClientIp,
   userAttributes: IAzureUserAttributes,
   servicePayload: ApiService
-) => Promise<IResponseSuccessJson<ApiService> | IResponseErrorQuery>;
+) => Promise<
+  | IResponseSuccessJson<ApiService>
+  | IResponseErrorQuery
+  | IResponseErrorValidation
+>;
 
 export function CreateServiceHandler(
   _GCTC: CustomTelemetryClientFactory,
   serviceModel: ServiceModel
 ): ICreateServiceHandler {
-  return async (_, __, ___, ____, servicePayload) => {
+  return async (context, __, ___, ____, servicePayload) => {
     const service = apiServiceToService(servicePayload);
     const errorOrCreatedService = await serviceModel.create(
       service,
       service.serviceId
     );
-    return errorOrCreatedService.fold<
-      IResponseErrorQuery | IResponseSuccessJson<ApiService>
-    >(
-      error => ResponseErrorQuery("CreateServiceHandler error", error),
-      createdService =>
-        ResponseSuccessJson(retrievedServiceToApiService(createdService))
+
+    if (isLeft(errorOrCreatedService)) {
+      return ResponseErrorQuery(
+        "CreateServiceHandler error",
+        errorOrCreatedService.value
+      );
+    }
+
+    const createdService = errorOrCreatedService.value;
+
+    const upsertServiceEvent = UpsertServiceEvent.encode({
+      newService: createdService,
+      updatedAt: new Date().getTime()
```

Comment:
we usually encode dates as UTC ISO strings, you should use [UTCISODateFromString](https://github.com/teamdigitale/io-ts-commons/blob/master/src/dates.ts#L41) in the definition of `UpsertServiceEvent` (instead of `t.integer`)

Reply:
Repository io-functions-app is using t.integer for events https://github.com/teamdigitale/io-functions-app/blob/master/utils/UpdatedProfileEvent.ts#L14.
I am going to change this PR to use UTCISODateFromString anyway.

Reply:
thanks for noticing, can you also fix the functions app? thanks